src/cockpit/channels/filesystem.py

```diff
@@ -160,6 +161,51 @@ def do_yield_data(self, options: JsonObject) -> Generator[bytes, None, JsonObjec
             raise ChannelError('internal-error', message=str(exc)) from exc
 
 
+class FSReplaceAttrs:
+    uid: 'int | None' = None
+    gid: 'int | None' = None
+    supported_attrs = ['owner', 'group']
+
+    def __init__(self, value: JsonObject) -> None:
+        # Any unknown keys throw an error
+        unsupported_attrs = list(value.keys() - self.supported_attrs)
+        if unsupported_attrs:
+            raise ChannelError('protocol-error',
+                               message=f'"attrs" contains unsupported key(s) {unsupported_attrs}',
+                               unsupported_attrs=unsupported_attrs) from None
+
+        try:
+            owner = get_str_or_int(value, 'owner', None)
+        except JsonError:
+            raise ChannelError('protocol-error', message='"owner" must be an integer or string') from None
```

Comment:
This is redundant.  The channel itself will raise the error if we hit a JsonError...  See https://github.com/cockpit-project/cockpit/commit/2a04612e0a6be26a3fbab3b6bbe512ffe12eb233

Reply:
ack